test/vscode-ui/test/specs/extension.spec.ts

```diff
@@ -0,0 +1,121 @@
+import { browser, $, expect } from "@wdio/globals";
+
+import * as fs from "fs";
+import * as path from "path";
+import { fileURLToPath } from "url";
+import { dirname } from "path";
+
+async function switchToSubframe() {
+  await browser.$(".webview");
+  const iframe = await browser.$("iframe");
+  await browser.switchToFrame(iframe);
+
+  await browser.$("iframe").waitForExist({ timeout: 30000 });
+  const subiframe = await browser.$("iframe");
+  await subiframe.waitForExist({ timeout: 30000 });
+  await browser.switchToFrame(subiframe);
+}
+
+function generateRandomString(length = 10) {
+  const characters =
+    "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
+  let result = "";
+  for (let i = 0; i < length; i++) {
+    result += characters.charAt(Math.floor(Math.random() * characters.length));
+  }
+  return result;
+}
+
+const serverName = generateRandomString();
+const apiKey = generateRandomString(32);
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = dirname(__filename);
+
+describe("VS Code Extension UI Test", () => {
+  let workbench: any;
+  let input: any;
+
+  before(async () => {
+    workbench = await browser.getWorkbench();
+    input = await $(".input");
+  });
+
+  it("open extension", async () => {
+    browser.$("aria/Posit Publisher").waitForExist({ timeout: 30000 });
```

Comment:
I am also always worried about timeouts.. did you set this one very high because of the unknown nature of our initialization? Could you loop until it appears instead?